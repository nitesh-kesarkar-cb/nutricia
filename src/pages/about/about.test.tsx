import { render, screen } from '@testing-library/react';
import AboutPage from './about'; // adjust the path/name if your file differs

describe('AboutPage', () => {
   it('renders the main heading', () => {
      render(<AboutPage />);
      const heading = screen.getByRole('heading', {
         level: 2,
         name: /about page/i,
      });
      expect(heading).toBeInTheDocument();
   });

   it('renders exactly one h2', () => {
      render(<AboutPage />);
      expect(screen.getAllByRole('heading', { level: 2 })).toHaveLength(1);
   });

   it('matches snapshot', () => {
      const { asFragment } = render(<AboutPage />);
      expect(asFragment()).toMatchSnapshot();
   });
});
