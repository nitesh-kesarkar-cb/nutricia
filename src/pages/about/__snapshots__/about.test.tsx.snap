// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutPage > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="prose dark:prose-invert max-w-none"
  >
    <h2
      class="text-3xl font-bold tracking-tight"
    >
      About Page
    </h2>
  </section>
</DocumentFragment>
`;
